cli/src/main/java/com/box/l10n/mojito/cli/command/checks/ContextAndCommentCliChecker.java

```diff
@@ -0,0 +1,117 @@
+package com.box.l10n.mojito.cli.command.checks;
+
+import com.box.l10n.mojito.cli.command.extraction.AssetExtractionDiff;
+import com.box.l10n.mojito.okapi.extractor.AssetExtractorTextUnit;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * {@link CliChecker} that verifies the comment and context parameters are provided and
+ * are not identical.
+ *
+ * @author mallen
+ */
+public class ContextAndCommentCliChecker extends AbstractCliChecker {
+
+    static Logger logger = LoggerFactory.getLogger(ContextAndCommentCliChecker.class);
+
+    class ContextAndCommentCliCheckerResult {
+        String sourceString;
+        String failureMessage;
+        boolean failed;
+
+        public ContextAndCommentCliCheckerResult(boolean failed, String sourceString, String failureMessage) {
+            this.sourceString = sourceString;
+            this.failureMessage = failureMessage;
+            this.failed = failed;
+        }
+
+        public ContextAndCommentCliCheckerResult(boolean failed) {
+            this.failed = failed;
+        }
+
+        public String getSourceString() {
+            return sourceString;
+        }
+
+        public String getFailureMessage() {
+            return failureMessage;
+        }
+
+        public boolean isFailed() {
+            return failed;
+        }
+    }
+
+    @Override
+    public CliCheckResult run() {
+        CliCheckResult cliCheckResult = new CliCheckResult(isHardFail(), CliCheckerType.CONTEXT_COMMENT_CHECKER.name());
+        StringBuilder notificationText = new StringBuilder();
+        runChecks(cliCheckResult, notificationText);
+        if(!cliCheckResult.isSuccessful()) {
+            cliCheckResult.setNotificationText("Context and comment check found failures:" + System.lineSeparator() + notificationText);
+        }
+        return cliCheckResult;
+    }
+
+    private void runChecks(CliCheckResult cliCheckResult, StringBuilder notificationText) {
+        getAddedTextUnits().stream()
```

Comment:
Is it possible to have the logic that run the check and then logic that builds the report separated? in theory we'd try to have avoid side effects in stream too